Remove bitstring dependency

greglucas opened the pull request:
# Title of PR

This removes the bitstring dependency and all bitstring inputs allowed with it. You can pass in the raw objects instead of the bitstring objects now.

~~This also removes the `legacy_parser` method which was kept around to support csv parsing. This has been removed in favor of converting the csv packet definitions into the XTCE format instead.~~

This sits on top of https://github.com/medley56/space_packet_parser/pull/58 so there wasn't as much thrashing going on. Only the second commit is the removal portion.

~~NOTE: I have removed the entire csvdef module. I am not sure if you want to keep that around or not. This is definitely the heavy-handed approach of just removing all of the legacy things and moving on in a major version bump.~~

## Checklist
- [x] Changes are fully implemented without dangling issues or TODO items
- [x] Deprecated/superseded code is removed or marked with deprecation warning
- [x] Current dependencies have been properly specified and old dependencies removed
- [x] New code/functionality has accompanying tests and any old tests have been updated to match any new assumptions
- [x] The changelog.md has been updated

medley56 commented:
Looks like there are some issues with the tests hanging around still. Some of the integration tests for CSV parsing we will either need to remove entirely or we will need to come up with an XTCE definition for the previously used CSV files.

greglucas commented:
> Looks like there are some issues with the tests hanging around still. Some of the integration tests for CSV parsing we will either need to remove entirely or we will need to come up with an XTCE definition for the previously used CSV files.

For some reason, I thought the csv parsing required bitstring which is why I removed it earlier. It turns out it doesn't, so I've now reverted all of the CSV changes and made this just a removal of bitstring now. This is much cleaner and can take place in two steps with a future removal of the legacy csv parser.

medley56 commented on docs/source/changelog.md:
```diff
@@ -6,7 +6,9 @@ list and release milestones.
 
 ### v5.0.0 (unreleased)
-- Replace bitstring objects with native Python bytes objects (except for in CSV-based parsing code)
+- Replace bitstring objects with native Python bytes objects
```
Leave a note in here that this is the breaking change and anyone using ConstBitStream objects as input to the generator will need to change to the (much prettier) direct input of a binary filelike object.

medley56 reviewed (approved):
This is awesome. Unless my recollection deceives me, the speed boost is perceptible in the integration tests.

medley56 commented on docs/source/changelog.md:
```diff
@@ -6,7 +6,9 @@ list and release milestones.
 
 ### v5.0.0 (unreleased)
-- Replace bitstring objects with native Python bytes objects (except for in CSV-based parsing code)
+- Replace bitstring objects with native Python bytes objects
+  - Remove dependency on the bitstring library
+  - Much faster parsing speed
```
```suggestion
- BREAKING: Replace `bitstring` objects with native Python bytes objects
  - Remove dependency on the `bitstring` library
  - Much faster parsing speed
  - Users that are passing `bitstring.ConstBitStream` objects to `generator` will need to pass a 
    binary filelike object instead
```
